_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,345 @@
+
+## Accessibility Support
+
+Different browsers have different levels of support for CSS. This can cause contrast issues in one browser that do not appear in another. Because of that, this rule can produce different results depending on the browser that is used. For example, a text that is positioned using CSS transform may be on a different background in a browser that does not support CSS transform.
+
+## Background
+
+- [Understanding Success Criterion 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
+- [G18: Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G18)
+- [G145: Ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G145)
+- [F83: Failure of Success Criterion 1.4.3 and 1.4.6 due to using background images that do not provide sufficient contrast with foreground text (or images of text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F83)
+- [CSS Scoping Module Level 1 (Editor's Draft)](https://drafts.csswg.org/css-scoping/)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This black [text node](https://dom.spec.whatwg.org/#text) is on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 2
+
+This black [text node](https://dom.spec.whatwg.org/#text) is on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F);">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 3
+
+This light [text node](https://dom.spec.whatwg.org/#text) is on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:25px; background: #000 no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Passed Example 4
+
+This [text node](https://dom.spec.whatwg.org/#text) passes because of a text shadow.
+
+```html
+<p style="color: #000; background: #737373; text-shadow: white 0 0 3px">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 5
+
+This text only has a 3:1 contrast but it is a large (18 point) text so it still passes.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 6
+
+This text only has a 3:1 contrast but it is a large (14 points and bold) text so it still passes.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+This text has an insufficient contrast but it is decorative so it still passes.
```

Comment:
But if you look at ACT-R glossary entry for decorative it reads "Serving only an aesthetic purpose, providing no information, and having no functionality."
I would argue that the text in this example meets this definition. But it would be nice to have decorative linked to its definition.

Reply:
@Brynanders The reason this text is decorative is because if this text was completely replaced with another sentence, nothing is lost. The purpose of this text is to demonstrate the font, any other sentence that contains all letters of the alphabet serves the same purpose. That means its decorative.

Reply:
I agree that  the meaning of the text is irrelevant here. But at the same time I feel a bit inconfortable saying that this can have very poor contrast and readability because
```html
<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
<p style="font-family: helvetica; background: #000; color: #001;" aria-hidden="true">
	The quick brown fox jumps over the lazy dog.
</p>
```
is not really demonstrating the font (I can't see what it look like if I can't see the text…)

Reply:
I agree with @Jym77 here. Though the text can be completely replaced with another sentence, it is also a visible example of the font type. One could argue that is not strictly only an aesthetic purpose or no functionality, because it demonstrates the font to the user.

Reply:
@Jym77 @DagfinnRomen  This aria-hidden block falls under 1.4.11: non-text contrast. It needs to meet 3:1, which in our example it does. Yes if it had a small contrast ratio, that would be a failure, but not of the SC we're testing in this rule.

Reply:
Hum, it's a bit hard to swallow that a text node is "non-text content", but I see your point and I agree with you.

It could be useful to expand a bit the description of the example to sum up this discussion (namely that the **meaning** is not important, hence it is not considered as text (it does not convey information) and thus is matched against 1.4.11, not 1.4.6).